includes/sanitizers/class-amp-meta-sanitizer.php

```diff
@@ -135,28 +135,18 @@ protected function ensure_viewport_is_present() {
 		 *
 		 * @var DOMElement $viewport_tag
 		 */
-		$viewport_tag      = $this->meta_tags[ self::TAG_VIEWPORT ][0];
-		$viewport_content  = $viewport_tag->getAttribute( 'content' );
-		$viewport_settings = array_map( 'trim', explode( ',', $viewport_content ) );
-		$width_found       = false;
-
-		// @todo The invalid/missing properties should raise validation errors. See <https://github.com/ampproject/amp-wp/pull/3758/files#r348074703>.
-		foreach ( $viewport_settings as $index => $viewport_setting ) {
-			list( $property, $value ) = array_map( 'trim', explode( '=', $viewport_setting ) );
-			if ( 'width' === $property ) {
-				if ( 'device-width' !== $value ) {
-					$viewport_settings[ $index ] = 'width=device-width';
-				}
-				$width_found = true;
-				break;
-			}
-		}
+		$viewport_tag     = $this->meta_tags[ self::TAG_VIEWPORT ][0];
+		$viewport_content = $viewport_tag->getAttribute( 'content' );
+
+		// Remove any whitespace and random delimiters (commas).
+		$viewport_settings = preg_replace( '/^,|,(?=[^A-Za-z0-9 ])|,$|\s+/', '', $viewport_content );
+		$width_found       = false !== strpos( $viewport_settings, 'width=' );
```

Comment:
I think the previous method for finding whether the `$width` was found is more robust. What if someone has a property like `body-width=foo`? This will incorrectly match here.

Reply:
Yep agreed. Fixed in https://github.com/ampproject/amp-wp/pull/4137/commits/4cde75a7766e388968506bfc4724abd924cdf8d0.